ProjFS.Mac/PrjFSLib/PrjFSLib.cpp

```diff
@@ -1120,10 +1124,13 @@ static bool GetXAttr(const char* fullPath, const char* name, size_t size, _Out_
     return false;
 }
 
-static bool RemoveXAttr(const char* fullPath, const char* name)
+static bool RemoveXAttr(const char* fullPath, const char* name, _Out_ int* error)
 {
+    *error = 0;
+    
     if (removexattr(fullPath, name, XATTR_NOFOLLOW))
```

Comment:
Apologies in advance for the newbie question:

Is the right definition for removexattr https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man2/removexattr.2.html
i.e. it returns an int 0 for success, -1 for failure?

Once I have my mac I can check myself!

Reply:
Thanks @nickgra for the verification!  I hear my mac is coming soon and I can't wait :-)
And that's what I was getting at, let's update the check.

Reply:
Thanks for the clarification @pmj!

Since the code is equivalent there isn't anything wrong here.  I am a fan of doing the "0 !=" as you suggest for clarity.  That said if the code already has multiple instances of this style, i'll leave it to @wilbaker to decide.

It's also a fair point to return errno_t.  Whatever we decide for this function perhaps we should consider for set/get as well.

Reply:
> Looking at the function as a whole though, I'm not quite clear on why we shouldn't just return an errno_t from the wrapper function directly instead of fiddling around with a boolean and an errno.

Two part answer:

- I went back and forth as to whether we should just return the errno.  Now that you're suggesting it as well that sounds like the better design 😄 

- I missed that we are using `errno_t` elsewhere in our code, so I'll make that change.  I went with `int` because I found that the headers define `errno` as an `int` rather than an `errno_t` (which I found surprising):

```
extern int * __error(void);
#define errno (*__error())
```

> Since the code is equivalent there isn't anything wrong here. I am a fan of doing the "0 !=" as you suggest for clarity. That said if the code already has multiple instances of this style, i'll leave it to @wilbaker to decide.

I am also not a fan of the implicit conversion so I'll make the change here and we can update other spots as we go.  (FWIW I'm also not a fan of our use of Yoda conditions, but I'm not sure where everyone else stands on that).

Reply:
I'm fine with the Yoda stuff if the left hand side is a very long expression (function call) so it's hard to see the comparison if it's all the way off to the right. I find it unnecessary on short expressions; the accidental assignment typo is caught by compilers these days anyway. (But I don't have strong feelings on this.)